biomage/utils/data.py

```diff
@@ -18,13 +77,57 @@ def modified_records(item, target_table, config):
             "meta": {
                 "M": {
                     **item["meta"]["M"],
-                    "pipeline": {
+                    "gem2s": {
                         "M": {
-                            "stateMachineArn": {"S": ""},
-                            "executionArn": {"S": ""},
+                            **item["meta"]["M"]["gem2s"]["M"],
+                            "paramsHash": {"S" : ""},
                         }
                     },
                 }
+            },
+            # Modify project id to point to sandboxed project
+            "projectId": {
+                'S': f"{extra['sandbox_id']}-{item['projectId']['S']}",
+            },
+            # Add the current user's id to experiment
+            **add_env_user_to_experiment(cfg={
+                "rbac_can_write": {
+                    "SS": item['rbac_can_write']['SS']
+                }
+            })
+        }
+
+    if target_table == config["staging-samples-table"]:
+        return {
+            "projectUuid": {"S" : f"{extra['sandbox_id']}-{item['projectUuid']['S']}"},
+            "samples": remap_sample_references(item['samples'], extra['sandbox_id'])
+        }
+
+    if target_table == config["staging-projects-table"]:
+        return {
+            "projects" : {
```

Comment:
same here, just modify the three fields you want to modify, it's completely unnecessary and inefficient to take the whole structure apart just to create it again in a new dict